testing.md

```diff
@@ -0,0 +1,196 @@
+- https://github.com/apache/cordova-android/blob/master/test/run_java_unit_tests.js
+- `npm run java-unit-tests`
+
+### Plugins
+
+Plugin tests are somewhat special, as they are set up in a different way. `npm run tests` just includes the syntax check, while the real tests are run by other means:
+
+#### Native
+
+https://github.com/apache/cordova-plugin-camera/tree/master/tests/ios
+https://github.com/apache/cordova-plugin-splashscreen/tree/master/tests/ios
+https://github.com/apache/cordova-plugin-wkwebview-engine/tree/master/tests/ios
+
+#### Plugin tests with `cordova-plugin-test-framework`
+
+The actual tests can be found as `test.js` in a `/tests` folder. The automated tests, exported as `defineAutoTests`, use the familiar `jasmine` syntax, while the manual tests can be any HTML elements - usually buttons that trigger a JS function, with a description of the expected result underneath - exported via `defineManualTests`.
+
+Additionally, the `/tests` folder contains its own `package.json` and `config.xml` to create a "mini test plugin" to be used by [`cordova-plugin-test-framework`](https://github.com/apache/cordova-plugin-test-framework) to create a test interface (manual UI and `jasmine` test runner) that is viewable inside an app: Create a new app, add the plugin, add the "tests" sub-plugin and you can navigate to `cdvtests/index.html` in your app to view that test interface.
+
+#### Appium Tests
+
+A small number of plugins also implements End to End Tests with Appium. They live in `/appium-tests`.
+TODO Native code is being execised vs. JS part with all the other test types
+
+#### Paramedic: `cordova-paramedic`
+
+To automate the creating of a test app and the test execution process [`cordova-paramedic`](https://github.com/apache/cordova-paramedic) can be be used:
+
+1. Install `cordova-paramedic` (`npm install -g ...` or clone + `npm install`)
+2. Run Paramedic for the current plugin:  
+    ```
+    node /tmp/paramedic/main.js 
+      --config pr/$PLATFORM 
+      --plugin $(pwd) 
+    ```
+3. App is built, setup; Tests are run and results reported
+
+##### Sauce Labs
+
+Paramedic can also run these tests on a remote service called Sauce Labs. They offer emulators and real devices of all kind to run testing on. Both normal plugin tests and Appium tests are executed on Sauce Labs on different operating systems and operating system versions.
+
+#### Core Plugins
+
+(Almost) All Core Plugins implement tests in the previously described ways. Some focus on automated tests, some on manual ones. The plugin tests are executed on CI by the already mentioned `cordova-paramedic` on Sauce Labs.
+
+#### 3rd party plugins
+
+TODO
+
+Some have tests, not very many unfortunately.
+
+
+## How to run tests?
+
+### Local
+
+As all Cordova projects are set up as JavaScript projects, the available test commands are usually defined in the `scripts` array of `package.json`. An example from `cordova-android`:
+
+```json
```

Comment:
If the point of this is to explain how to find commands to run, an alternative is simply to run `npm run` without a subcommand.

```
$ npm run
Lifecycle scripts included in cordova-android:
  test
    npm run eslint && npm run cover && npm run java-unit-tests

available via `npm run-script`:
  unit-tests
    jasmine --config=spec/unit/jasmine.json
  cover
    nyc jasmine --config=spec/coverage.json
  e2e-tests
    jasmine --config=spec/e2e/jasmine.json
  java-unit-tests
    node test/run_java_unit_tests.js
  eslint
    eslint bin spec test
```

It's nicely formatted, showing the subcommand alias, the commands that the alias runs, and the user doesn't need to read though the `package.json`.

If the point also includes how potential more subcommand can be added and where, then what is written is OK.

Just  alternative information.